sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -75,9 +85,43 @@ public DefaultService(ServiceSpec serviceSpec, Collection<Plan> plans) {
         register(serviceSpec, this.plans);
     }
 
+    private void lock() {
+        String rootPath = CuratorUtils.toServiceRootPath(serviceSpec.getName());
+        String lockPath = CuratorUtils.join(rootPath, "LOCK_PATH");
+        curatorMutex = new InterProcessMutex(curatorClient, lockPath);
+
+        LOGGER.info("Acquiring ZK lock...");
+        try {
+            if (!curatorMutex.acquire(10, TimeUnit.SECONDS)) {
+                LOGGER.error("Failed to acquire ZK lock.  Exiting.");
+                System.exit(1);
+            }
+        } catch (Exception ex) {
+            LOGGER.error("Error acquiring ZK lock.", ex);
+        }
+    }
+
+    private void unlock() {
+        try {
+            curatorMutex.release();
+        } catch (Exception ex) {
+            LOGGER.error("Error releasing ZK lock.", ex);
```

Comment:
Oh whoops it looks like `getLockPath()` is `protected`. Nevermind. Worst case someone can scroll back to when `lock()` was called to figure out the path.